tests/unit/states/test_slack.py

```diff
@@ -47,29 +48,115 @@ def test_post_message(self):
             comt = ('The following message is to be sent to Slack: {0}'
                     .format(message))
             ret.update({'comment': comt})
-            self.assertDictEqual(slack.post_message(name, channel, from_name,
-                                                    message), ret)
+            self.assertDictEqual(slack.post_message(name,
+                                                    channel=channel,
+                                                    from_name=from_name,
+                                                    message=message,
+                                                    api_key=api_key), ret)
 
         with patch.dict(slack.__opts__, {'test': False}):
-            comt = ('Slack channel is missing: None')
+            comt = ('Please specify api_key or webhook.')
             ret.update({'comment': comt, 'result': False})
-            self.assertDictEqual(slack.post_message(name, None, from_name,
-                                                    message), ret)
+            self.assertDictEqual(slack.post_message(name,
+                                                    channel=None,
+                                                    from_name=from_name,
+                                                    message=message,
+                                                    api_key=None), ret)
 
-            comt = ('Slack from name is missing: None')
+            comt = ('Slack channel is missing.')
             ret.update({'comment': comt, 'result': False})
-            self.assertDictEqual(slack.post_message(name, channel, None,
-                                                    message), ret)
+            self.assertDictEqual(slack.post_message(name,
+                                                    channel=None,
+                                                    from_name=from_name,
+                                                    message=message,
+                                                    api_key=api_key), ret)
 
-            comt = ('Slack message is missing: None')
+            comt = ('Slack from name is missing.')
```

Comment:
Instead of same as above, any string which fits in a single line and yet has `()` around it. Let's remove the `()`.